src/lib.rs

```diff
@@ -922,6 +959,34 @@ impl ConfigFile {
     }))
   }
 
+  pub fn to_publish_config(&self) -> Result<Option<PublishConfig>, AnyError> {
+    let files_config = self.to_files_config()?;
+    let publish_config = match self.json.publish.clone() {
+      Some(config) => {
+        let publish_config: SerializedPublishConfig =
+          serde_json::from_value(config)
+            .context("Failed to parse \"test\" configuration")?;
+        Some(publish_config.into_resolved(&self.specifier)?)
+      }
+      None => None,
+    };
+
+    if files_config.is_none() && publish_config.is_none() {
+      return Ok(None);
+    }
+
+    Ok(Some(match publish_config {
+      Some(publish_config) => match files_config {
+        Some(files_config) => publish_config.with_files(files_config),
+        None => publish_config,
+      },
+      None => PublishConfig {
+        files: files_config
+          .unwrap_or_else(|| FilePatterns::new_with_base(self.dir_path())),
+      },
+    }))
+  }
```

Comment:
Not sure if this is handled, but should there be a way to opt-out of a top level exclude? Should this work?

```json
{
  "publish: {
    "include": "./dist"
  },
  "exclude": ["./dist"]
}
```

Someone may want to exclude the dist folder from the lsp, fmt, lint, etc, but then include it for publishing because it contains a build of the package.

Reply:
Seems complicated. This uses the same infra as other command level excludes. Can we fix this in a follow up?

Reply:
Yup, I think it's the only situation where there's a use case. Could you open an issue when you get a chance?

Reply:
https://github.com/denoland/deno_config/pull/34#discussion_r1463726264